src/containers/CreateAdministrator/index.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Create Administrator
+ */
+
+import React, { useState } from "react";
+import { axiosInstance } from "../../config/axios";
+import AdministratorForm from "../../components/AdministradorForm";
+import propTypes from "prop-types";
+
+CreateAdministrator.propTypes = {
+  setNotify: propTypes.func.isRequired,
+};
+
+export default function CreateAdministrator({ setNotify }) {
+  const [administrator, setAdministrator] = useState({
+    email: "",
+    first_name: "",
+    last_name: "",
+    password: "",
+    password_confirmation: "",
+  });
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (administrator.email == "" || administrator.password == "" || administrator.password_confirmation == "") {
+      setNotify({
+        isOpen: true,
+        message: "Falta completar campos obligatorios.",
+        type: "error",
+        reload: false,
+      })
+    } else if (administrator.password != administrator.password_confirmation) {
+      setNotify({
+        isOpen: true,
+        message: "Las contraseñas no coinciden",
+        type: "error",
+        reload: false,
+      });
+    } else {
+        axiosInstance
+          .post("/users", {
+            user: {
+              ...administrator,
+            },
+          })
+          .then((response) => {
+            if (response.status == 200)
+              setNotify({
+                isOpen: true,
+                message: `El administrador se creo con exito.`,
+                type: "success",
+                reload: true,
+              });
+            else
+              setNotify({
+                isOpen: true,
+                message: `Error inesperado.`,
+                type: "error",
+                reload: false,
+              });
+          })
+          .catch((error) => {
+            let errorsArray = error.response.data.errors.full_messages;
+            let message = "";
+            errorsArray.forEach((e) => (message += e + ", "));
+            setNotify({
+              isOpen: true,
+              message: message.slice(0, -2),
+              type: "error",
+              reload: false,
+            });
+          });
+      };
+  }
+
+  const handleInputChange = (e) => {
+    if (e.target.id == "email") {
+      setAdministrator({
+        ...administrator,
+        email: e.target.value.toLowerCase().replace(" ", ""),
+      });
+    } else if (e.target.id == "first_name") {
+      setAdministrator({ ...administrator, first_name: e.target.value });
+    } else if (e.target.id == "last_name") {
+      setAdministrator({ ...administrator, last_name: e.target.value });
+    } else if (e.target.id == "password") {
+      setAdministrator({ ...administrator, password: e.target.value });
+    } else if (e.target.id == "password_confirmation") {
+      setAdministrator({ ...administrator, password_confirmation: e.target.value });
+    }
```

Comment:
```suggestion
    setAdministrator((prev)=> ({...prev, [e.target.id]: e.target.value});
```
Incluso queda mejor si es con `e.target.name` porque hace referencia al key del name y en el `<InputField name="email" ... />` por ejemplo

Reply:
demas, ahi lo cambie. No entedi lo del name asi que lo deje con id. Decime que te parece asi

` const handleInputChange = (e) => {
    let value = e.target.value;
    if (e.target.id == "email") { //esto capaz no es necesario y en la linea del set dejamos: e.target.value
      value = e.target.value.toLowerCase().replace(" ", "");
    }
    setAdministrator((prev)=> ({...prev, [e.target.id]: value}));
    if (
      (e.target.id == "password" || e.target.id == "password_confirmation") &&
      !(administrator.password == administrator.password_confirmation)
    )
      setNotify({
        isOpen: true,
        message: "Las contraseñas no coinciden",
        type: "error",
        reload: false,
      });
  }; `

Reply:
Excelente, lo del name es porque hace referencia a la key, el id podria ser 'person-form-email'. Pero es lo mismo, dejalo asi.
```
if (e.target.id == "email") { //esto capaz no es necesario y en la linea del set dejamos: e.target.value
      value = e.target.value.toLowerCase().replace(" ", "");
}
```
Esto se podría sacar si, el backend va a validar que sea un email.
`let value = e.target.value;` se podria cambiar por `const value = e.target.value;`